scripts/post-install/post-install.js

```diff
@@ -0,0 +1,40 @@
+const fs = require('fs');
+const Logger = require('../utils/logger');
+
+/**
+ * Validates the installation of terra-toolkit.
+ * Logs a warning if terra-toolkit is detected as a hard dependency.
+ * terra-toolkit is expected to always be installed as a devDependency.
+ */
+function validateInstall() {
+  /**
+   * INIT_CWD returns the initial working directory of the npm install.
+   * See the following issue for more information: https://github.com/npm/npm/issues/16990
+   * Note: This feature was introduced in npm 5.4.0.
+   */
+  if (!process.env.INIT_CWD) {
+    Logger.warn('WARNING: [terra-toolkit] - INIT_CWD was undefined. This is likely due to an outdated version of npm. Please consider upgrading. terra-toolkit does not support this version of npm.');
```

Comment:
We just now decided with this file change that we don't support npm prior to version 5.4.0. I'm not sure this checker is a good enough reason to claim we don't support an npm version.

Reply:
There's a discussion here regarding the change: https://github.com/cerner/terra-toolkit/pull/329#discussion_r322816362

I'm fine with removing it or keeping it. The definition of our support is a little blurry with our declaration here: https://github.com/cerner/terra-toolkit/blob/master/package.json#L7

Reply:
Node and NPM versions are tightly coupled: https://nodejs.org/en/download/releases/

Reply:
Removed the unsupported message: https://github.com/cerner/terra-toolkit/pull/329/commits/43e0f8877f28d110e342f0caa1f8d1fc4d0bbbf1